File: libs/gomodels/channel_chat_message.go
package model

import "time"

type ChannelChatMessage struct {
	ID           uint      `gorm:"primary_key;column:id;type:serial"         json:"id"`
	MessageId    string    `gorm:"primary_key;column:message_id;type:TEXT;" json:"messageId"`
	ChannelId    string    `gorm:"column:channel_id;type:text"              json:"channelId"`
	UserId       string    `gorm:"column:user_id;type:text"                 json:"userId"`
	UserName     string    `gorm:"column:user_name;type:text"               json:"userName"`
	Text         string    `gorm:"column:text;type:text"                   json:"text"`
	CanBeDeleted bool      `gorm:"column:can_be_deleted;type:bool"           json:"canBeDeleted"`
	CreatedAt    time.Time `gorm:"column:created_at;type:timestamp"         json:"createdAt"`

	Channel *Channels `gorm:"foreignKey:ID" json:"channel"`
	User    *Users    `gorm:"foreignKey:ID" json:"user"`
}

func (c *ChannelChatMessage) TableName() string {
	return "channels_messages"
}

File: libs/typeorm/src/entities/ChannelChatMessage.ts
/* eslint-disable import/no-cycle */
import {
  Column,
  CreateDateColumn,
  Entity,
  Index,
  JoinColumn,
  ManyToOne,
  PrimaryColumn,
} from 'typeorm';

import { Channel } from './Channel';
import { User } from './User';

@Entity('channels_messages')
export class ChannelChatMessage {
  @PrimaryColumn('text')
  messageId: string;

  @Column('text')
  @Index()
  channelId: string;

  @ManyToOne(() => Channel, _ => _.messages)
  @JoinColumn({ name: 'channelId' })
  channel?: Channel;

  @Column('text')
  @Index()
  userId: string;

  @ManyToOne(() => User, _ => _.messages)
  @JoinColumn({ name: 'userId' })
  user?: User;

  @Column('text')
  userName: string;

  @Column('text')
  text: string;

  @Column('boolean', { default: true })
  canBeDeleted: boolean;

  @CreateDateColumn()
  createdAt: Date;
}
